### ISSUE TITLE
Some improvements suggested on the documentation

### ISSUE THREAD

**Issue by thomas-touhey** [jepler/leapseconddata#18]
Hi, the project looks nice and the code looks really tidy, but the project is missing a documentation right about now. Would you be interested in a Sphinx documentation, that you could probably host on Read the Docs or any other static platform? :)

(plus, the project has an easy link from Pypi to Github, but it would also be great to have a link from Github to Pypi, to verify authenticity)

**Comment by jepler** [comment]
Thanks for offering to help out!

There are docs at https://leapseconddata.readthedocs.io/en/latest/ but they are not particularly complete, and the build was recently failing. I have corrected this. There also aren't any docs for tagged releases, just "latest".

There are badge-style links from readthedocs & github README.md to pypi. This is a style I'm accustomed to, but not everyone is. Is there something not working about them, or is the problem insufficient discoverability?

![image](https://github.com/user-attachments/assets/c44882f3-e1d4-4014-801d-0a64baf1bdaa)

If you feel like helping out more generally, I noticed that this repo is still using pyproject.toml+setup.cfg. I'd like it to use pyproject.toml only. PRs welcome. I haven't thought much of other opportunities for contribution.

**Comment by thomas-touhey** [comment]
Hi,

So I somehow skipped the docs/ directory because there was no link on neither the PyPI package nor the README, and am kind of used just to skip the badges because they don't usually contain the info I'm looking for. All of this is my bad, although I guess some more explicit links within the text and an additional link on PyPI would help out? For example, "flask" has a "Documentation" link: https://pypi.org/project/Flask/

As for the existing docs, it's great that it exists and is linked to the RtD platform already, but it's only a reference for now. It would actually be great to have at least two more components from [Diataxis](https://diataxis.fr/), which are the topics, e.g.:

- What are leap seconds? Who defines them, how are they interacting with concepts like TAI, UT1, UTC? Would probably mostly be composed of links such as [The UNIX leap second mess](http://www.madore.org/%7Edavid/computers/unix-leap-seconds.html), [Bulletin C](https://datacenter.iers.org/productMetadata.php?id=16), and so on.
- How is this information actually distributed? This could define the rationale behind the sources, i.e. the rationales behind your "from_" methods. So for example:

  - For the file, who produces the `/usr/share/zoneinfo/leap*` files, which packages, is the path standardized and in which standard, how often should the file be updated (every 6 months, around probably Jan or Feb, then July or Aug, depending on the distribution
  - For the Github URL or meinberg.de URL, why trust these sources and not others, such as [the online equivalent of the Debian package you already trust as a file source](https://salsa.debian.org/glibc-team/tzdata)?

Then the second component could be guides, as actual recipes on what "user stories", or actual problems, you are answering to. In the use cases I'm planning to use this module to, it would for example be, in a "Developer guides" section:

- Converting a TAI date and time to an UTC date and time
- Converting a UTC date and time to a TAI date and time
- Checking if a :60 date and time is valid / if a minute is a leap minute
- Keeping the leap second data up-to-date in a long-running application (I'm using [cachetools](https://cachetools.readthedocs.io/en/latest/) for that, but it doesn't fully answer my problem for that since it doesn't support an expiry date returned by the result unfortunately)

It's mostly just ideas (and fuelled by, as described earlier, a WIP project where I'm choosing to depend on this one to actually handle dates and times for now), feel free to tell me no at any point o_o

**Comment by jepler** [comment]
It's a cliche that open source documentation could almost always be better.

I've updated my packaging metadata so that a "documentation" link is called out on the left sidebar of pypi (in prerelease 4.1.0-alpha.1):

![image](https://github.com/user-attachments/assets/f7940617-5343-4438-943e-dd3f5b053407)

Out of your specific suggestions, writing up something about how I chose the leap second data sources is the one I'm most likely to do (well, more to the point, having forgotten the history at this point, writing a justification for what's there: "I web-searched and found these leap second sources")

If you are inclined to contribute documentation, I will be happy to review it. A little can probably go a long way.
